### PR TITLE
Add GP SDFs/namelists for HR3

### PR DESCRIPTION
**Pull request by dustinswales**
This PR adds RRTMGP enabled HR3 SDFs.
Also, rename namelist from input_GFS_v17_H3 -> input_GFS_v17_H**R**3

### CODE REVIEW

**Comment by mzhangw** [comment]
Per @lisa-bengtsson, for HR3, we may want to set "progsigma=.T." in *nml and use the the field_table with sigmab. @lisa-bengtsson has shared the SCM configs with HR1 and HR2 with @dustinswales, but not with HR3.

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  do_gsl_drag_ss       = .true.
+  do_gsl_drag_tofd     = .true.
+  do_ugwp_v1_orog_only = .false.
+  min_lakeice  = 0.15
+  min_seaice   = 0.15
+  use_cice_alb = .false.
+  pdfcld       = .false.
+  fhswr        = 1200.
+  fhlwr        = 1200.
+  ialb         = 1
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:44]: Per the setting in RTs, it looks like ialb/iems=2 is more convincing for GFS suite with Noahmp.  We may need to double-check it.

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  iems         = 1
+  iaer         = 5111
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:46]: 1011

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  print_diff_pgr = .false.
+  fhcyc        = 24
+  use_ufo      = .true.
+  pre_rad      = .false.
+  imp_physics  = 8
+  iovr         = 3
+  ltaerosol    = .false.
+  lradar       = .true.
+  ttendlim     = -999
+  dt_inner     = 225.0
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:15]: 150

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+  do_ugwp_v0   = .false.
+  do_ugwp_v1   = .true.
+  do_ugwp_v0_orog_only = .false.
+  do_ugwp_v0_nst_only  = .false.
+  do_gsl_drag_ls_bl    = .true.
+  do_gsl_drag_ss       = .true.
+  do_gsl_drag_tofd     = .true.
+  do_ugwp_v1_orog_only = .false.
+  min_lakeice  = 0.15
+  min_seaice   = 0.15
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:39]: 1.0e-6

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+  use_cice_alb = .false.
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:40]: .true.

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+  pdfcld       = .false.
+  fhswr        = 1200.
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:42]: 3600.

Comment (mzhangw) [reply ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:42]: 3600

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+  fhlwr        = 1200.
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:43]: 3600.

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml (hunk 1 above)
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:44]: 2

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+  ialb         = 1
+  iems         = 1
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:45]: 2

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+  swhtr        = .true.
+  cnvgwd       = .true.
+  shal_cnv     = .true.
+  cal_pre      = .false.
+  redrag       = .true.
+  dspheat      = .true.
+  hybedmf      = .false.
+  satmedmf     = .true.
+  isatmedmf    = 1
+  lheatstrg    = .true.
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:62]: .false.

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+  lseaspray    = .true.
+  random_clds  = .false.
+  trans_trac   = .true.
+  cnvcld       = .true.
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:66]: need add progsigma=.true.

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+  imfshalcnv   = 2
+  imfdeepcnv   = 2
+  ras          = .false.
+  cdmbgwd      = 10.0,3.5,1.0,1.0
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:70]: 2.5,7.5,1.0,1.0  for C786
1.67,8.8,1.0,1.0 for C1152

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+  iopt_crs     = 2
+  iopt_btr     = 1
+  iopt_run     = 1
+  iopt_sfc     = 3
+  iopt_trs     = 2
+  iopt_frz     = 1
+  iopt_inf     = 1
+  iopt_rad     = 3
+  iopt_alb     = 1
+  iopt_snf     = 4
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:86]: need to check if 4 the default value

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+  iopt_tbot    = 2
+  iopt_stc     = 3
+  debug        = .false.
+  nstf_name    = 2,1,0,0,0
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:90]: 2,0,0,0

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+  nst_anl      = .true.
+  psautco      = 0.0008,0.0005
+  prautco      = 0.00015,0.00015
+  lgfdlmprad   = .false.
+  effr_in      = .true.
+  ldiag_ugwp   = .false.
+  fscav_aero   = "'*:0.0'"
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:97]: delete

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+  lw_file_clouds = 'extensions/cloud_optics/rrtmgp-cloud-optics-coeffs-lw.nc'
+  sw_file_gas = 'rrtmgp/data/rrtmgp-data-sw-g112-210809.nc'
+  sw_file_clouds = 'extensions/cloud_optics/rrtmgp-cloud-optics-coeffs-reordered-sw.nc'
+  rrtmgp_nGptsSW = 112
+  rrtmgp_nGptsLW = 128
+  rrtmgp_nBandsLW = 16
+  rrtmgp_nBandsSW = 14
+  frac_grid    = .true.
+  cplchm       = .false.
+  cplflx       = .false.
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:117]: true

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 16]
```diff
@@ -0,0 +1,171 @@
...
+  cplice       = .false.
+  cplwav       = .false.
+  cplwav2atm   = .false.
+  do_ca        = .false.
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:121]: True

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 17]
```diff
@@ -0,0 +1,171 @@
...
+  ca_sgs       = .false.
+  nca          = 1
+  ncells       = 5
+  nlives       = 12
+  nseed        = 1
+  nfracseed    = 0.5
+  nthresh      = 18
+  ca_trigger   = .true.
+  nspinup      = 1
+  iseed_ca     = 12345
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:132]: 1448371824 for C786
1437371824 for C1152

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 18]
```diff
@@ -0,0 +1,171 @@
...
+  lndp_type    = 0
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:133]: delete

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 19]
```diff
@@ -0,0 +1,171 @@
...
+  n_var_lndp   = 0
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:134]: delete

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 20]
```diff
@@ -0,0 +1,171 @@
...
+  do_ccpp_suite_sim = .false.
+/
+
+&cires_ugwp_nml
+  knob_ugwp_solver   = 2
+  knob_ugwp_version  = 1
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:140]: delete

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 21]
```diff
@@ -0,0 +1,171 @@
...
+  knob_ugwp_wvspec   = 1,25,25,25
+  knob_ugwp_azdir    = 2,4,4,4
+  knob_ugwp_stoch    = 0,0,0,0
+  knob_ugwp_effac    = 1,1,1,1
+  knob_ugwp_doaxyz   = 1
+  knob_ugwp_doheat   = 1
+  knob_ugwp_dokdis   = 2
+  knob_ugwp_ndx4lh   = 4
+  knob_ugwp_palaunch = 275.0e2
+  knob_ugwp_nslope   = 0
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:151]: 1

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 22]
```diff
@@ -0,0 +1,171 @@
...
+  knob_ugwp_lzmax    = 15.750e3
+  knob_ugwp_lzmin    = 0.75e3
+  knob_ugwp_lzstar   = 2.0e3
+  knob_ugwp_taumin   = 0.25e-3
+  knob_ugwp_tauamp   = 1.5e-3
+  knob_ugwp_lhmet    = 200.0e3
+  knob_ugwp_orosolv  = 'pss-1986'
+/
+
+&ccpp_suite_sim_nml
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:161]: do we need this section?

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 23]
```diff
@@ -0,0 +1,171 @@
+&gfs_physics_nml
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:1]: The changes I mentioned below are based on @lisa-bengtsson and Jiande's HR3 configuration. There are also applied to input_GFS_v17_HR3.nml

**Review by mzhangw — CHANGES_REQUESTED** [review]

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml [hunk 24]
```diff
@@ -0,0 +1,171 @@
...
+  knob_ugwp_doheat   = 1
+  knob_ugwp_dokdis   = 2
+  knob_ugwp_ndx4lh   = 4
+  knob_ugwp_palaunch = 275.0e2
+  knob_ugwp_nslope   = 0
+  knob_ugwp_lzmax    = 15.750e3
+  knob_ugwp_lzmin    = 0.75e3
+  knob_ugwp_lzstar   = 2.0e3
+  knob_ugwp_taumin   = 0.25e-3
+  knob_ugwp_tauamp   = 1.5e-3
```
Comment (mzhangw) [inline ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:156]: 0.5e-3 for C786 and 0.35e-3 for C1152

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml (hunk 23 above)
Comment (dustinswales) [reply ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:1]: @mzhangw The GP nml is identical to the HR3 SDF, with one exception, do_RRTMGP to True.

Can you open a PR into this branch with the updated input_GFS_v17_HR3.nml from @lisa-bengtsson? And I will update the GP counterpart in here.

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml (hunk 24 above)
Comment (dustinswales) [reply ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:156]: I wasn't planning on changing the resolutions from the HR3 SDF to the HR3_RRTMGP SDF. So whatever resolution you decided on is fine by me.

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml (hunk 22 above)
Comment (dustinswales) [reply ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:161]: yes

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml (hunk 23 above)
Comment (mzhangw) [reply ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:1]: will do.

File: ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml (hunk 16 above)
Comment (mzhangw) [reply ccpp/physics_namelists/input_GFS_v17_HR3_RRTMGP.nml:121]: actually off for SCM

**Comment by dustinswales** [comment]
@mzhangw @grantfirl Closing this PR. I'm including the GP changes on top of Man's HR3 PR, #474

**Pull request closed without merging**